templates/docker/Dockerfile

```diff
@@ -84,6 +84,11 @@ RUN /src/k8s-snap/build-scripts/build-component.sh containerd
 FROM builder AS build-helm
 RUN /src/k8s-snap/build-scripts/build-component.sh helm
 
+# # kubernetes build with older version for in place upgrade testing
+# FROM builder AS build-kubernetes-old
+# RUN echo "$(cat /src/k8s-snap/build-scripts/components/kubernetes/version | cut -f-2 -d'.').0" > /src/k8s-snap/build-scripts/components/kubernetes/version
+# RUN /src/k8s-snap/build-scripts/build-component.sh kubernetes
+
```

Comment:
we have (or will soon have) a `KUBERNETES_VERSION` arg to choose which Kubernetes version to build. Let's add a `KUBERNETES_VERSION_UPGRADE_TO`. If set, this builds an extra binary and adds it to our well-known location